app/src/main/java/com/example/iou/bill/activities/SplitSettlementActivity.java

```diff
@@ -63,7 +65,7 @@ protected void onCreate(Bundle savedInstanceState) {
         StringBuilder str = new StringBuilder();
         for (Iterator i = keys.iterator(); i.hasNext(); ) {
```

Comment:
For readability and maintainability, I think its easier to use `for (String key : amountsOwed.keySet())` instead of using iterators. It's also a less breakable as you don't have to call `i.next()`